### ISSUE TITLE
jsTree not been created when tree-data="scope"

### ISSUE THREAD

**Issue by liujoey** [arvindr21/jsTree-directive#13]
when set tree-data="scope", the jsTree will not be initiated unless the scope data changes.

in the code you setup the watch, but the function will not be triggered until I call $scope.$apply() some where.

```
s.$watch(a.treeModel, function(o, n) {
    if (n) {
        config = {
        'core': {
            'data': s[a.treeModel]
        }
        };
        $(e).jstree('destroy');
        treeDir.init(s, e, a, config);
    }
}, true);
```

**Comment by liujoey** [comment]
please also update the tag and bower file. Currently bower installing a very old version.

**Comment by arvindr21** [comment]
Fixed. Please check. Also pushed the latest code to bower https://github.com/arvindr21/jsTree-directive/releases/tag/v0.1.3

**Comment by arvindr21** [comment]
Can you confirm if the updated fix works as expected? I can release a new bower version.

**Comment by liujoey** [comment]
I left a comment on your last commit, but it seems github doesn't show it anywhere unless you open the commit itself. Well, please allow me copy the comment here.
```
+          // Trigger it initally
+          // Fix issue #13
+          treeDir.init(s, e, a, config);
```
the three lines you added here, seems the config object was not assigned to a correct value.

**Comment by arvindr21** [comment]
I did see the comment and fixed it. Reopened the issue again, so we can track. Ref : https://github.com/arvindr21/jsTree-directive/commit/1efa94041d6e7b8d70f070852f64a3bc314f7080

**Comment by liujoey** [comment]
tested, and it's working, thank you for the quick response.

**Comment by arvindr21** [comment]
Thanks! You use bower version [`0.1.4`](https://github.com/arvindr21/jsTree-directive/releases/tag/v0.1.4).

**Comment by IllyaMoskvin** [comment]
Apologies for revisiting this issue, but I had some trouble with jsTree v3.0.9, which I fixed by commenting out the full `// Trigger it initally` code of this fix. Namely, I had the following error thrown:

Callstack | File
------------ | -------------
Uncaught TypeError: Cannot read property 'triggerHandler' of null  | jstree.js:857
$.jstree.core.trigger  | jstree.js:857
$.jstree.core._append_json_data.rslt  | jstree.js:1773
(anonymous function) | jstree.js:1802 
n.extend.proxy.n.isFunction.e | jquery.min.js:2 

I'm using a `$resource` factory to get data from my API. For reference, the relevant parts of my code look somewhat like this:

    angular.isUndefinedOrNull = function(val) {
        return angular.isUndefined(val) || val === null 
    }

    app.filter('jstree', function() {
        return function( array ) {
            return array.map( function( e, i, a ) {
                
                // jsTree represents root nodes with parent = #
                e.parent = e.category_id == 0 ? '#' : e.category_id;
                delete e.category_id;
                
                // See 'Alternative JSON Format'
                // http://www.jstree.com/docs/json/
                e.text = e.title;
                delete e.title;
                
                return e;
                
            });
        }
    });

    $scope.sidebarTree = [ { id: 1, parent: '#', text: 'Loading...' } ];

    Category.get({}, function(data) {
        $scope.categories = data.items;
    });

    $scope.$watch( 'categories', function (newVal, oldVal) {
        if( !angular.isUndefinedOrNull( newVal ) ) {
            $scope.sidebarTree = $filter('jstree')($scope.categories);
        }
    });

    <js-tree tree-data="scope" tree-model="sidebarTree" ></js-tree>

Basically, I think that `jsTree.directive` does not play nice when AJAX calls are used to build the model for `tree-data="scope"`. Maybe the issue is on my end, but I wanted to bring this to your attention just in case it's something deeper. 

Sorry for the snippet-heavy comment. Please let me know if I should clarify any info.

**Comment by arvindr21** [comment]
@IllyaMoskvin Thanks. Can you try the below

```js
$scope.$watch( 'categories', function (newVal, oldVal) {
    console.log(newVal, oldVal);
    if( !angular.isUndefinedOrNull( newVal ) ) {
        var _tree = $filter('jstree')($scope.categories);
        console.log(_tree);
        $scope.sidebarTree = _tree;
    }
});
```
and share the output here?

Thanks.

**Comment by IllyaMoskvin** [comment]
Certainly. Thanks for the quick response! I truncated the existing table to create a simplified test case and verified that the bug still occurs. I also wrapped each variable inside the `console.log()` calls in a `JSON.stringify()` to make copying easier and to avoid running into the "Object state below is captured on first expansion" problem with Chrome. Here is the output:

    undefined undefined
    [{"id":1,"category_id":0,"title":"Root Node"},{"id":2,"category_id":1,"title":"Foo"},{"id":3,"category_id":1,"title":"Bar"}] undefined
    [{"id":1,"parent":"#","text":"Root Node"},{"id":2,"parent":1,"text":"Foo"},{"id":3,"parent":1,"text":"Bar"}]
    Uncaught TypeError: Cannot read property 'triggerHandler' of null

I hope this helps! Thanks.

**Comment by meriturva** [comment]
The same here with versione 0.1.5. The error is always:
`Uncaught TypeError: Cannot read property 'triggerHandler' of null`

I have fixed removed the line `// Trigger it initally` mentioned by @IllyaMoskvin
